src/app/content/highlights/hooks/locationChange.ts

```diff
@@ -1,18 +1,22 @@
 import { GetHighlightsSourceTypeEnum } from '@openstax/highlighter/dist/api';
+import { getType } from 'typesafe-actions';
+import { receivePageFocus } from '../../../actions';
 import { user } from '../../../auth/selectors';
-import { AppServices, MiddlewareAPI } from '../../../types';
+import { AnyAction, AppServices, MiddlewareAPI } from '../../../types';
 import { bookAndPage } from '../../selectors';
 import { receiveHighlights } from '../actions';
 import * as select from '../selectors';
 
-const hookBody = (services: MiddlewareAPI & AppServices) => async() => {
+const hookBody = (services: MiddlewareAPI & AppServices) => async(action?: AnyAction) => {
   const {dispatch, getState, highlightClient} = services;
   const state = getState();
   const {book, page} = bookAndPage(state);
   const authenticated = user(state);
   const loaded = select.highlightsLoaded(state);
 
-  if (!authenticated || !book || !page || typeof(window) === 'undefined' || loaded) {
+  const pageFocusIn = action && action.payload && action.type === getType(receivePageFocus);
```

Comment:
This is throwing me an error:
```
Property 'payload' does not exist on type 'EmptyAction<"Auth/receiveLoggedOut">'
```
You can probably just change it to:
```
  const pageFocusIn = action && action.type === getType(receivePageFocus) && action.payload;
```

Reply:
@KarinaMendez2 pretty sure piotr is right about this change